### DISCUSSION TITLE
Consider moving Starlight config to separate file

### DISCUSSION THREAD

**Discussion by tribal2** [mearashadowfax/ScrewFast#45] [Ideas]
I noticed the project's config is quite extensive, largely due to Starlight settings. What are your thoughts on extracting these into a separate starlight.config.mjs file for better readability and maintainability? If you're on board, I'd be happy to work on this adjustment.

**Comment by mearashadowfax** [comment]
That would be great! I totally agree that the project's config has become quite extensive, especially with the addition of docs. Extracting the Starlight settings into a separate file would definitely improve readability and maintainability. Thank you for taking the initiative to work on this adjustment! Additionally, we might consider changing the social image format to .webp from .png in Starlight's settings, I forgot to do it, when we updated Meta component.

**Comment by tribal2** [comment]
I'm also working in auto-generating og images for individual pages.. I'll submit a PR when I have this working

**Comment by newbeelearn** [comment]
wow dynamic og generation if og image is not supplied would be great addition. Please add additional parameter ogimgtitle if possible for auto generating og image if ogimgtitle is provided and than fallback to post title. This will give additional flexibility. In my blog i auto generate og iamge but sometimes title does not looks good when og image is generated.

**Comment by tribal2** [comment]
@newbeelearn see #31
